core/src/main/java/com/coderanch/blackjack/Hand.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright (C) 2018 Coderanch.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.coderanch.blackjack;
+
+import java.util.*;
+
+import static com.coderanch.util.require.Require.requireThat;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.notNullValue;
+
+/**
+ * A hand in a game of Blackjack.
+ */
+final class Hand {
+
+    /**
+     * Highest score in Blackjack.
+     */
+    private static final int MAX_LEGAL_SCORE = 21;
+
+    /**
+     * Max amount of aces
+     */
+    private static final int MAX_SCORE_COUNT = 4;
+
+
+    private List<Card> cards;
+    private List<Integer> possibleScores;
+    private int bestScore;
+
+    /**
+     * Creates a new hand.
+     *
+     * @param card  first card dealt.
+     * @param card2 second card dealt
+     * @throws NullPointerException if either {@code card} or {@code card2} is {@code null}.
+     */
+    public Hand(Card card, Card card2) {
+        this(List.of(card, card2));
```

Comment:
```suggestion
    public Hand(Card firstCard, Card secondCard) {
        requireThat("firstCard",  firstCard,  is(notNullValue()));
        requireThat("secondCard", secondCard, is(notNullValue()));
        this(List.of(firstCard, secondCard));
```

Reply:
I originally wanted to write it like this.  It won't compile because `this()` has to be called first.